x-pack/plugins/triggers_actions_ui/public/application/components/rules_setting/rules_settings_modal.tsx

```diff
@@ -0,0 +1,299 @@
+
+  const { isLoading, isError: hasError } = useGetFlappingSettings({
+    enabled: isVisible,
+    onSuccess: (fetchedSettings) => {
+      if (!settings) {
+        setSettings({
+          enabled: fetchedSettings.enabled,
+          lookBackWindow: fetchedSettings.lookBackWindow,
+          statusChangeThreshold: fetchedSettings.statusChangeThreshold,
+        });
+      }
+    },
+  });
+
+  const { mutate } = useUpdateFlappingSettings({
+    onSave,
+    onClose,
+    setUpdatingRulesSettings,
+  });
+
+  // In the future when we have more settings sub-features, we should
+  // disassociate the rule settings capabilities (save, show) from the
+  // sub-feature capabilities (writeXSettingsUI).
+  const canWriteFlappingSettings = save && writeFlappingSettingsUI && !hasError;
+  const canShowFlappingSettings = show && readFlappingSettingsUI;
+
+  const handleSettingsChange = (
+    key: keyof RulesSettingsFlappingProperties,
+    value: number | boolean
+  ) => {
+    if (!settings) {
+      return;
+    }
+
+    const newSettings = {
+      ...settings,
+      [key]: value,
+    };
+
+    setSettings({
+      ...newSettings,
+      statusChangeThreshold: Math.min(
+        newSettings.lookBackWindow,
+        newSettings.statusChangeThreshold
+      ),
+    });
+  };
+
+  const handleSave = () => {
+    if (!settings) {
+      return;
+    }
+    mutate(settings);
+  };
+
+  if (!isVisible) {
+    return null;
+  }
+
+  const maybeRenderForm = () => {
+    if (hasError || !canShowFlappingSettings) {
+      return <RulesSettingsErrorPrompt />;
+    }
+    if (!settings || isLoading) {
+      return <CenterJustifiedSpinner />;
+    }
+    return (
+      <EuiForm>
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <RulesSettingsFlappingTitle />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiSpacer size="s" />
+        <EuiFlexGroup>
+          <RulesSettingsModalFormLeft
+            isSwitchDisabled={!canWriteFlappingSettings}
+            settings={settings}
+            onChange={(e) => handleSettingsChange('enabled', e.target.checked)}
+          />
+          <RulesSettingsModalFormRight
+            settings={settings}
+            onChange={(key, value) => handleSettingsChange(key, value)}
+          />
+        </EuiFlexGroup>
+      </EuiForm>
+    );
+  };
+
+  return (
+    <EuiModal data-test-subj="rulesSettingsModal" onClose={onClose} maxWidth={880}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <h3>
+            <FormattedMessage
+              id="xpack.triggersActionsUI.rulesSettings.modal.title"
+              defaultMessage="Rule Settings"
+            />
+          </h3>
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiCallOut
+          size="s"
+          title={i18n.translate('xpack.triggersActionsUI.rulesSettings.modal.calloutMessage', {
+            defaultMessage: 'Applies to all rules within the current space',
```

Comment:
Since "rule settings" is a plural, IMO this should be "Apply" or "Applicable":
```suggestion
            defaultMessage: 'Apply to all rules within the current space.',
```

Reply:
Addressed in https://github.com/elastic/kibana/pull/149086